From 17751f15c2d15782ee24ddcfe991de940dbdc3b0 Mon Sep 17 00:00:00 2001
From: Thomas Lamprecht <t.lamprecht@proxmox.com>
Date: Wed, 6 Apr 2016 13:28:16 +0200
Subject: [PATCH 46/55] pve: cleanup includes all over the place

QEMU wants to get a cleaner include system, adapt to it.
---
 block/zeroinit.c |  3 +++
 savevm-async.c   |  4 ++++
 vma-reader.c     | 11 +----------
 vma-writer.c     | 12 +++---------
 vma.c            |  9 +--------
 5 files changed, 12 insertions(+), 27 deletions(-)

diff --git a/block/zeroinit.c b/block/zeroinit.c
index 04a9bd0..eb77657 100644
--- a/block/zeroinit.c
+++ b/block/zeroinit.c
@@ -8,9 +8,12 @@
  * See the COPYING file in the top-level directory.
  */
 
+#include "qemu/osdep.h"
+#include "qapi/error.h"
 #include "block/block_int.h"
 #include "qapi/qmp/qdict.h"
 #include "qapi/qmp/qstring.h"
+#include "qemu/cutils.h"
 
 typedef struct {
     bool has_zero_init;
diff --git a/savevm-async.c b/savevm-async.c
index bb4372c..4d73e66 100644
--- a/savevm-async.c
+++ b/savevm-async.c
@@ -1,3 +1,4 @@
+#include "qemu/osdep.h"
 #include "qemu-common.h"
 #include "qapi/qmp/qerror.h"
 #include "qemu/error-report.h"
@@ -13,6 +14,9 @@
 #include "qemu/timer.h"
 #include "sysemu/block-backend.h"
 #include "qapi/qmp/qstring.h"
+#include "qemu/rcu.h"
+#include "qemu/thread.h"
+#include "qemu/cutils.h"
 
 /* #define DEBUG_SAVEVM_STATE */
 
diff --git a/vma-reader.c b/vma-reader.c
index 5d0d3ea..cb58360 100644
--- a/vma-reader.c
+++ b/vma-reader.c
@@ -11,14 +11,7 @@
  *
  */
 
-#include <stdio.h>
-#include <errno.h>
-#include <unistd.h>
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <fcntl.h>
+#include "qemu/osdep.h"
 #include <glib.h>
 #include <uuid/uuid.h>
 
@@ -28,8 +21,6 @@
 #include "vma.h"
 #include "block/block.h"
 
-#define BITS_PER_LONG  (sizeof(unsigned long) * CHAR_BIT)
-
 static unsigned char zero_vma_block[VMA_BLOCK_SIZE];
 
 typedef struct VmaRestoreState {
diff --git a/vma-writer.c b/vma-writer.c
index 3cde204..e96fb55 100644
--- a/vma-writer.c
+++ b/vma-writer.c
@@ -11,22 +11,16 @@
  *
  */
 
-#include <stdio.h>
-#include <errno.h>
-#include <unistd.h>
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <fcntl.h>
+#include "qemu/osdep.h"
 #include <glib.h>
 #include <uuid/uuid.h>
 
-#include "qemu-common.h"
 #include "vma.h"
 #include "block/block.h"
 #include "monitor/monitor.h"
 #include "qemu/main-loop.h"
+#include "qemu/coroutine.h"
+#include "qemu/cutils.h"
 
 #define DEBUG_VMA 0
 
diff --git a/vma.c b/vma.c
index 46ae14b..c3db633 100644
--- a/vma.c
+++ b/vma.c
@@ -11,14 +11,7 @@
  *
  */
 
-#include <stdio.h>
-#include <errno.h>
-#include <unistd.h>
-#include <stdio.h>
-#include <string.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <fcntl.h>
+#include "qemu/osdep.h"
 #include <glib.h>
 
 #include "vma.h"
-- 
2.1.4

